src/data/entities/D2ExpressionParser.ts

```diff
@@ -0,0 +1,114 @@
+    boolean: xp.ValueType.BOOLEAN,
+    date: xp.ValueType.DATE,
+    number: xp.ValueType.NUMBER,
+};
+
+export class D2ExpressionParser {
+    public evaluateRuleEngineCondition(
+        ruleCondtion: string,
+        ruleVariables: Map<ProgramRuleVariableName, ProgramRuleVariableValue>
+    ): Either<Error, boolean> {
+        try {
+            const expressionParser = new xp.ExpressionJs(
+                ruleCondtion,
+                xp.ExpressionMode.RULE_ENGINE_CONDITION
+            );
+
+            const variables = expressionParser.collectProgramRuleVariableNames();
+            const variablesValueMap = this.mapProgramVariables(variables, ruleVariables);
+            const variablesMap = new Map(
+                variablesValueMap.map(variable => [variable.programRuleVariable, variable.value])
+            );
+
+            const programVariables = expressionParser.collectProgramVariablesNames();
+            programVariables.forEach(programVariable => {
+                switch (programVariable) {
+                    case "current_date": {
+                        variablesMap.set(
+                            programVariable,
+                            this.getVariableValueByType(
+                                "date",
+                                new Date().toISOString().split("T")[0]
+                            )
+                        );
+                        break;
+                    }
+                    default:
+                        throw new Error(
+                            `Unhandled Program variable of type : ${programVariable}. Please contact developer`
+                        );
+                }
+            });
+
+            const expressionData = new xp.ExpressionDataJs(
+                variablesMap,
+                undefined,
+                undefined,
+                undefined,
+                undefined
+            );
+
+            const parsedResult: boolean = expressionParser.evaluate(
+                () => console.debug(""),
+                expressionData
+            );
+
+            return Either.success(parsedResult);
+        } catch (error) {
+            return Either.error(
+                new Error(
+                    `An error occurred while evaluating the rule in D2ExpressionParser::evaluateRuleEngineCondition: ${error}`
+                )
+            );
+        }
+    }
+
+    private getVariableValueByType = (
+        type: ProgramRuleVariableType,
+        stringValue: xp.Nullable<string>
+    ): xp.VariableValueJs => {
+        const valueType = VariableValueTypeMap[type];
+        return new xp.VariableValueJs(valueType, stringValue, [], null);
+    };
+
+    private mapProgramVariables(
+        programRuleVariables: string[],
+        ruleVariables: Map<string, ProgramRuleVariableValue>
+    ) {
+        return programRuleVariables.map(programRuleVariable => {
+            const currentProgramRuleVariableValue = ruleVariables.get(programRuleVariable);
```

Comment:
[subjective] short-lived variables (where the context is clear) can have shorter names.